config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecWithBcpTest.java

```diff
@@ -0,0 +1,254 @@
+import com.yahoo.yolean.Exceptions;
+import org.junit.Test;
+
+import java.io.StringReader;
+import java.time.Duration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * @author bratseth
+ */
+public class DeploymentSpecWithBcpTest {
+
+    @Test
+    public void minimalProductionSpecWithExplicitBcp() {
+        StringReader r = new StringReader("""
+                <deployment version='1.0'>
+                   <instance id='default'>
+                      <prod>
+                         <region>us-east1</region>
+                         <region>us-west1</region>
+                      </prod>
+                   </instance>
+                   <bcp>
+                     <group>
+                       <region>us-east1</region>
+                       <region>us-west1</region>
+                     </group>
+                   </bcp>
+                </deployment>
+            """);
+        assertTwoRegions(DeploymentSpec.fromXml(r));
+    }
+
+    @Test
+    public void specWithoutInstanceWithBcp() {
+        StringReader r = new StringReader("""
+                <deployment version='1.0'>
+                   <prod>
+                      <region>us-east1</region>
+                      <region>us-west1</region>
+                   </prod>
+                   <bcp>
+                     <group>
+                       <region>us-east1</region>
+                       <region>us-west1</region>
+                     </group>
+                   </bcp>
+                </deployment>
+            """);
+        assertTwoRegions(DeploymentSpec.fromXml(r));
+    }
+
+    @Test
+    public void complexBcpSetup() {
+        StringReader r = new StringReader("""
+                <deployment version='1.0'>
+                   <instance id='beta'>
+                      <prod>
+                         <region>us-east1</region>
+                         <region>us-east2</region>
+                      </prod>
+                      <bcp>
+                         <group deadline="60m">
+                            <region>us-east1</region>
+                            <region>us-east2</region>
+                         </group>
+                      </bcp>
+                   </instance>
+                   <instance id='main'>
+                      <prod>
+                         <region>us-east1</region>
+                         <region>us-east2</region>
+                         <region>us-central1</region>
+                         <region>us-west1</region>
+                         <region>us-west2</region>
+                         <region>eu-east1</region>
+                         <region>eu-west1</region>
+                      </prod>
+                   </instance>
+                   <bcp>
+                     <group>
+                        <region>us-east1</region>
+                        <region>us-east2</region>
+                        <region fraction="0.3">us-central1</region>
```

Comment:
I assume we'll discuss the syntax later. Maybe `traffic-share` or simply `share` would be more descriptive?

Reply:
Yes, on Thursday. This isn't traffic share though, but fractional membership ...